### DISCUSSION TITLE
Deprecate `hostType` and add `platformType` for host rules

### DISCUSSION THREAD

**Discussion by viceice** [renovatebot/renovate#11495] [Suggest an Idea]
The current `hostType` behavior is pretty invisible. It's mainly used to filter hostrules and set by datasource.

So we have the idea to fully deprecate / remove that config option and instead add a new `platformType` property, which can be used to define a specific type to the host rule.

With this information renovate can apply platform specific auth header more easily.

Sample:
```json
{
  "matchHost": "src.my-domain.test",
  "platformType": "gitlab",
  "token": "12345678901234567890"
}
```

With this rule renovate knows that it must use gitlab auth header and because the token has a length of 20 char's it need to use the `Private-token: 12345678901234567890` header instead of normal `Authorization: Bearer 12345678901234567890`  header.

/cc @rarkins @JamieMagee @HonkingGoose

**Comment by rarkins** [comment]
I would prefer to use the field `authType` to influence the auth header.

What I'm considering is: is it reasonable to insist on a hostname/domain name/url for each host rule? I think that a hostType-only authentication host rule is risky because it could send the auth to wrong hosts.

Then that leaves the use case of matchHost + hostType today. In most cases the hostType is redundant, because there's only one token per host. In cases where there may be different auth for different datasources, then using URLs for `matchHost` should work?

**Comment by LeoniePhiline** [comment]
A change like this would make stuff like https://github.com/renovatebot/renovate/issues/10649 less likely to happen. 

While creating the fix https://github.com/renovatebot/renovate/pull/14160 I got the impression that the credentials system might benefit from a general refactor.
